chingu-x/chingu-connect
# New/#54 frontend

closes #54 

<img width="894" alt="screen shot 2018-05-11 at 5 22 55 pm" src="https://user-images.githubusercontent.com/25065329/39947471-082b2352-5540-11e8-8871-ea62a53d5f9b.png">


## Purpose
* Create frontend code architecture
* Connect backend to frontend
* Implement apollo graphql on frontend

## Approach
* Setup webpack configuration
  * Includes common, development, and production configurations
  * Includes babel setup, SASS, HMR, and a few optimizations for production build
* Added react plugin for ESLint
* Created a `client` directory for all frontend code
  * Includes a `src` directory for development and a `dist` directory for production
    * `dist` directory is ignored when pushing up to github
* Setup SASS in the `src` directory
  * New .scss files need to be pre-fixed with an underscore (ie: `_newfile.scss`)
  * New .scss files need to be imported in `main.scss` (order matters)
* Included `concurrently` to simultaneously run webpack-dev-server with express server
* Added new scripts to `package.json`:
  * `npm run build:dev` runs the dev build for webpack and is served at `localhost:8080`
  * `npm run build:prod` creates `dist` directory with prod build for webpack
  * `npm run hack` runs frontend webpack dev server simultaneously with backend express server
    * Runs both scripts: `npm run build:dev` and `npm run test:server`

## Testing
* Download PR to local machine
* Run `npm install` to grab new dependencies
* For quick start, run `npm run hack` and go to `localhost:8080` to see app
